packages/core/admin/admin/src/content-manager/hooks/useRelation/useRelation.js

```diff
@@ -125,6 +125,19 @@ export const useRelation = (cacheKey, { relation, search }) => {
 
   const onLoadRelationsCallback = useCallbackRef(onLoadRelations);
 
+  const clearRelationsCallback = useCallbackRef(clearRelations);
+
+  /**
+   * This fires on unmount only.
+   */
+  useEffect(() => {
+    return () => {
+      if (clearRelationsCallback) {
```

Comment:
The idea looks good to me - good thinking. What do you think about renaming the callback to `onUnmount`, though?